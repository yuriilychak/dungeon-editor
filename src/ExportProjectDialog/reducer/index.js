import STATE from "../state";
import StaticData from "../../data/ExportProjectDialogData";

export const initialState = {
    staticData: StaticData,
    isPopupOpen: false
};

const actionHandlers = {
    [STATE.CHANGE_ACTIVITY]: (state, action) => {
        return {
            ...state,
            isPopupOpen: action.payload
        }
    }
};

export default function topMenuReducer(state = initialState, action) {
    const actionHandler = actionHandlers[action.type];
    return actionHandler ? actionHandler(state, action) : state;
}
