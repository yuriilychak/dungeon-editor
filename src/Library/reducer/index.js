import StaticData from "../data/index";
import STATE from "../state";

/**
 * @typedef {Object}
 * @name LibraryData
 * @property {number} id
 * @property {string} locale
 */

/**
 * @typedef {Object}
 * @name LibraryState
 * @property {Array<LibraryData>} tabs
 * @property {Array<Object>} files
 */

/**
 * @type {LibraryState}
 */

export const initialState = {
    ...StaticData,
    files: [[], [], [], [], []]
};

/**
 * @type {Object.<string, function(LibraryState, ActionData): LibraryState>}
 */

const actionHandlers = {
    [STATE.ADD_FILE]: (state, action) => {
        const files = state.files.slice(0);
        const {sectionId, data} = action.payload;
        const file = {
            title: data.name,
            id: data.id,
            hasPreview: false,
            isDirectory: false
        };
        files[sectionId] = [...files[sectionId], file];
        return {
            ...state,
            files
        }
    },
    [STATE.REMOVE_FILE]: (state, action) => {
        const files = state.files.slice(0);
        const index = action.payload.sectionId;
        files[index] = files[index].filter(file => file.id !== action.payload.id);
        return {
            ...state,
            files
        }
    },
    [STATE.UPDATE_TREE]: (state, action) => {
        const files = state.files.slice(0);
        const {fileTree, sectionId} = action.payload;
        files[sectionId] = fileTree;
        return {...state, files};
    },
    [STATE.ADD_DIRECTORY]: (state, action) => {
        const files = state.files.slice(0);
        const {sectionId, dirData} = action.payload;
        const fileTree = files[sectionId].slice(0);
        const parentId = dirData.parentId;
        const result = {
            title: dirData.name,
            id: dirData.id,
            hasPreview: false,
            isDirectory: true
        };

        if (parentId === -1) {
            fileTree.push(result);
        }
        else {
            const parent = findDir(fileTree, parentId);

            if (!parent) {
                return state;
            }

            if (!parent.children) {
                parent.children = [];
            }

            parent.children.push(result);
        }

        files[sectionId] = fileTree;

        return {...state, files};
    },
    [STATE.CLEAR]: state => ({...state, files: [[], [], [], [], []]})
};

function findDir(data, id) {
    const dirs = data.filter(element => element.isDirectory);
    let result = dirs.find( dir => dir.id === id);

    if (result) {
        return result;
    }

    dirs.forEach( dir => {
        if (!dir.children || result) {
            return;
        }

        result = findDir(dir.children, id);
    });

    return result;
}

/**
 * @param {LibraryState} state
 * @param {ActionData} action
 * @return {LibraryState}
 */

export default function topMenuReducer(state = initialState, action) {
    const actionHandler = actionHandlers[action.type];
    return actionHandler ? actionHandler(state, action) : state;
}
