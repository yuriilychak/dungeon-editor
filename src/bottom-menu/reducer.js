import StaticData from "./data";
import STATE from "./state";
export const initialState = {
    ...StaticData,
    sectionId: -1,
    fileId: -1,
    tabIndex: 0,
    elements: []
};

const actionHandlers = {
    [STATE.OPEN_ELEMENT]: (state, { fileId, sectionId }) => {
        return {
            ...state,
            fileId,
            sectionId,
            tabIndex: 0
        }
    },
    [STATE.CHANGE_TAB]: (state, tabIndex) => ({ ...state, tabIndex })
};

export default function topMenuReducer(state = initialState, { type, payload}) {
    const actionHandler = actionHandlers[type];
    return actionHandler ? actionHandler(state, payload) : state;
}

