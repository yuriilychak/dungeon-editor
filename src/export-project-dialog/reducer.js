import STATE from "./state";
import {generateReducerData} from "../helpers";

/**
 * @typedef {Object}
 * @name ProgressData
 * @property {number} progress
 * @property {string} fileName
 * @property {boolean} isComplete
 */

/**
 * @typedef {Object}
 * @name ExportProjectDialogState
 * @property {Object.<string, string>} locale
 * @property {boolean} isPopupOpen
 * @property {?ProgressData} progressData
 */


/**
 * @param {ExportProjectDialogState} state
 * @param {ActionData} action
 * @return {ExportProjectDialogState}
 */

export default generateReducerData(
    {
        isPopupOpen: false,
        progressData: null
    },
    {
        [STATE.CHANGE_ACTIVITY]: (state, isPopupOpen) => ({
            ...state,
            isPopupOpen,
            progressData: null
        }),
        [STATE.PROGRESS]: (state, progressData) => ({ ...state, progressData })
    }
);
