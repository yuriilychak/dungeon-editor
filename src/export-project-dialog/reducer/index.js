import STATE from "../state";
import StaticData from "../data";

/**
 * @typedef {Object}
 * @name ProgressData
 * @property {number} progress
 * @property {string} fileName
 * @property {boolean} isComplete
 */

/**
 * @typedef {Object}
 * @name ExportProjectDialogState
 * @property {Object.<string, string>} locale
 * @property {boolean} isPopupOpen
 * @property {?ProgressData} progressData
 */

/**
 * @type {ExportProjectDialogState}
 */

export const initialState = {
    ...StaticData,
    isPopupOpen: false,
    progressData: null
};

/**
 * @type {Object.<string, function(ExportProjectDialogState, ActionData): ExportProjectDialogState>}
 */

const actionHandlers = {
    [STATE.CHANGE_ACTIVITY]: (state, action) => ({
        ...state,
        isPopupOpen: action.payload,
        progressData: null
    }),
    [STATE.PROGRESS]: (state, action) => ({
        ...state,
        progressData: action.payload
    })

};

/**
 * @param {ExportProjectDialogState} state
 * @param {ActionData} action
 * @return {ExportProjectDialogState}
 */

export default function topMenuReducer(state = initialState, action) {
    const actionHandler = actionHandlers[action.type];
    return actionHandler ? actionHandler(state, action) : state;
}
