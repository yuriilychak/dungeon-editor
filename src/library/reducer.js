import StaticData from "./data";
import STATE from "./state";
import { handleAction } from "../helpers";

/**
 * @typedef {Object}
 * @name LibraryData
 * @property {number} id
 * @property {string} locale
 */

/**
 * @typedef {Object}
 * @name LibraryState
 * @property {Array<LibraryData>} tabs
 * @property {Array<Object>} files
 */

/**
 * @type {LibraryState}
 */

export const initialState = {
    ...StaticData,
    files: [[], [], [], [], [], [], []]
};

/**
 * @type {Object.<string, function(LibraryState, ActionData): LibraryState>}
 */

const actionHandlers = {
    [STATE.ADD_DIRECTORY]: (state, payload) => addElementToTree(payload, true, state),
    [STATE.ADD_FILE]: (state, payload) => addElementToTree(payload, false, state),
    [STATE.REMOVE_FILE]: (state, payload) => {
        const files = state.files.slice(0);
        const { sectionId, id } = payload;
        const fileTree = files[sectionId];
        const element = findElement(fileTree, id);
        const parentId = element.parentId;

        if (parentId === -1) {
            files[sectionId] = fileTree.filter(element => element.id !== id);
        }
        else {
            const parent = findElement(fileTree, parentId);
            parent.children = parent.children.filter(element => element.id !== id);

            files[sectionId] = fileTree.slice(0);
        }

        return {
            ...state,
            files
        }
    },
    [STATE.UPDATE_TREE]: (state, payload) => {
        const files = state.files.slice(0);
        const {fileTree, sectionId} = payload;
        refreshParentIds(fileTree);

        files[sectionId] = fileTree;

        return {...state, files};
    },
    [STATE.RENAME_FILE]: (state, payload) => {
        const files = state.files.slice(0);
        const {id, sectionId, name} = payload;
        const fileTree = files[sectionId];

        const element = findElement(fileTree, id);

        if (!element) {
            return state;
        }

        element.title = name;

        files[sectionId] = fileTree.slice(0);

        return {...state, files};
    },
    [STATE.CLEAR]: state => ({...state, files: [[], [], [], [], [], [], []]})
};

function refreshParentIds(files, parentId = -1) {
    files.forEach(file => {
        file.parentId = parentId;
        if (file.isDirectory && file.children && file.children.length !== 0) {
            refreshParentIds(file.children, file.id);
        }
    })
}

function addElementToTree(payload, isDirectory, state) {
    const files = state.files.slice(0);
    const {sectionId, data} = payload;
    const file = {
        title: data.name,
        id: data.id,
        parentId: data.parentId,
        preview: data.preview,
        isDirectory: isDirectory
    };
    const parentId = file.parentId;
    const fileTree = files[sectionId].slice(0);

    if (parentId === -1) {
        fileTree.push(file);
    }
    else {
        const parent = findElement(fileTree, parentId);

        if (!parent) {
            return state;
        }

        if (!parent.children) {
            parent.children = [];
        }

        parent.children.push(file);
    }

    files[sectionId] = fileTree;

    return {
        ...state,
        files
    };
}

function findElement(data, id) {
    const dirs = data.filter(element => element.isDirectory);
    let result = data.find(element => element.id === id);

    if (result) {
        return result;
    }

    dirs.forEach( dir => {
        if (!dir.children || result) {
            return;
        }

        result = findElement(dir.children, id);
    });

    return result;
}

/**
 * @param {LibraryState} state
 * @param {ActionData} action
 * @return {LibraryState}
 */

export default function topMenuReducer(state = initialState, action) {
    return handleAction(state, actionHandlers, action);
}
