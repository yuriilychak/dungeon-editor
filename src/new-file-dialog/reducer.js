import STATE from "./state";
import {generateReducerData} from "../helpers";

const generateResultState = (
    state,
    isPopupOpen = false,
    sectionId = -1,
    elementType = -1
) => ({
    ...state,
    sectionId,
    elementType,
    isPopupOpen
});

export default generateReducerData(
    {
        sectionId: -1,
        elementType: -1,
        isPopupOpen: false
    },
    {
        [STATE.OPEN_POPUP]: (state, sectionId) => {
            const { sections } = state;
            const section = sections.find(element => element.id === sectionId);

            return generateResultState(state, true, sectionId, section.types[0].id);
        },
        [STATE.CLOSE_POPUP]: state => generateResultState(state),
        [STATE.CHANGE_TYPE]: (state, elementType) => ({ ...state, elementType })
    }
);

