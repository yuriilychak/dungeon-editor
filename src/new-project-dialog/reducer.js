import STATE from "./state";
import StaticData from "./data/index.json";

export const initialState = {
    staticData: StaticData,
    isPopupOpen: false
};

const actionHandlers = {
    [STATE.CHANGE_ACTIVITY]: (state, action) => {
        return {
            ...state,
            isPopupOpen: action.payload
        }
    }
};

export default function topMenuReducer(state = initialState, action) {
    const actionHandler = actionHandlers[action.type];
    return actionHandler ? actionHandler(state, action) : state;
}

