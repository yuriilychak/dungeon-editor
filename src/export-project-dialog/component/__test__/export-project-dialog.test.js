import React from "react";
import { createMount } from "@material-ui/core/test-utils";

import { initialState } from '../../reducer';
import ExportPopupDialog from "../export-project-dialog";

describe("new-project-dialog index test",()=> {
    const props = {
        ...initialState,
        isPopupOpen: true,
        onClosePopup: jest.fn(),
        onExportProject: jest.fn()
    };

    const wrapper = createMount()(
        <ExportPopupDialog
            {...props}
        />
    );

    it('Open snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Close snapshot', () => {
        const innerWrapper = createMount()(
            <ExportPopupDialog
                {...initialState}
                isPopupOpen={false}
                onClosePopup={ jest.fn() }
                onExportProject={ jest.fn() }
            />
        );
        expect(innerWrapper.html()).toMatchSnapshot();
    });

    it("Test progress 0", () => {
        wrapper.setProps({
            progressData: {
                progress: 0,
                fileName: null,
                isComplete: false
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Test progress 20", () => {
        wrapper.setProps({
            progressData: {
                progress: 20,
                fileName: "test.json",
                isComplete: false
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Test progress 100 not completed", () => {
        wrapper.setProps({
            progressData: {
                progress: 100,
                fileName: null,
                isComplete: false
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Test progress 100 completed", () => {
        wrapper.setProps({
            progressData: {
                progress: 100,
                fileName: null,
                isComplete: true
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });
});
