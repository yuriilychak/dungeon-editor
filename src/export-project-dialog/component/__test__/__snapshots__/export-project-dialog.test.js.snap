// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-project-dialog index test Close snapshot 1`] = `""`;

exports[`export-project-dialog index test Open snapshot 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper action-dialog-container MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\" aria-labelledby=\\"export-project-title\\"><div class=\\"MuiDialogTitle-root\\" id=\\"export-project-title\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Export project</h2></div><div class=\\"MuiDialogContent-root\\"><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Wait until generate zip.</p><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Wait wheal project export start</p><div class=\\"MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate\\" role=\\"progressbar\\" aria-valuenow=\\"0\\"><div class=\\"MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate\\" style=\\"transform: translateX(-100%);\\"></div></div></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button><span class=\\"MuiButton-label\\">Cancel</span></button><button><span class=\\"MuiButton-label\\">Save</span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;

exports[`export-project-dialog index test Test progress 0 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper action-dialog-container MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\" aria-labelledby=\\"export-project-title\\"><div class=\\"MuiDialogTitle-root\\" id=\\"export-project-title\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Export project</h2></div><div class=\\"MuiDialogContent-root\\"><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Wait until generate zip.</p><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Add file null to archive</p><div class=\\"MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate\\" role=\\"progressbar\\" aria-valuenow=\\"0\\"><div class=\\"MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate\\" style=\\"transform: translateX(-100%);\\"></div></div></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button><span class=\\"MuiButton-label\\">Cancel</span></button><button><span class=\\"MuiButton-label\\">Save</span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;

exports[`export-project-dialog index test Test progress 20 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper action-dialog-container MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\" aria-labelledby=\\"export-project-title\\"><div class=\\"MuiDialogTitle-root\\" id=\\"export-project-title\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Export project</h2></div><div class=\\"MuiDialogContent-root\\"><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Wait until generate zip.</p><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Add file test.json to archive</p><div class=\\"MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate\\" role=\\"progressbar\\" aria-valuenow=\\"20\\"><div class=\\"MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate\\" style=\\"transform: translateX(-80%);\\"></div></div></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button><span class=\\"MuiButton-label\\">Cancel</span></button><button><span class=\\"MuiButton-label\\">Save</span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;

exports[`export-project-dialog index test Test progress 100 completed 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper action-dialog-container MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\" aria-labelledby=\\"export-project-title\\"><div class=\\"MuiDialogTitle-root\\" id=\\"export-project-title\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Export project</h2></div><div class=\\"MuiDialogContent-root\\"><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Wait until generate zip.</p><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Click save for continue</p><div class=\\"MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate\\" role=\\"progressbar\\" aria-valuenow=\\"100\\"><div class=\\"MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate\\" style=\\"transform: translateX(0%);\\"></div></div></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button><span class=\\"MuiButton-label\\">Cancel</span></button><button><span class=\\"MuiButton-label\\">Save</span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;

exports[`export-project-dialog index test Test progress 100 not completed 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper action-dialog-container MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\" aria-labelledby=\\"export-project-title\\"><div class=\\"MuiDialogTitle-root\\" id=\\"export-project-title\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Export project</h2></div><div class=\\"MuiDialogContent-root\\"><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Wait until generate zip.</p><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Wait wheal archive generate</p><div class=\\"MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate\\" role=\\"progressbar\\" aria-valuenow=\\"99\\"><div class=\\"MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate\\" style=\\"transform: translateX(-1%);\\"></div></div></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button><span class=\\"MuiButton-label\\">Cancel</span></button><button><span class=\\"MuiButton-label\\">Save</span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
